.github/workflows/static-analysis.yml

```diff
@@ -38,7 +38,7 @@ jobs:
           - locked
     steps:
       - name: 📦 Check out the codebase
-        uses: actions/checkout@v4.1.5
```

Comment:
Better to stick to full versions and use renovatebot or dependabot to do automatic CI upgrades. This is considered as good practice.